### ISSUE TITLE
sfml/2.5.1@bincrafters/stable: Link error xcb_get_setup, error adding symbols: DSO missing from command line

### ISSUE THREAD

**Issue by charmyt** [bincrafters/community#1236]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **sfml/2.5.1@bincrafters/stable**
  * Operating System+version: **Linux Ubuntu 19.10**
  * Compiler+version: **GCC 9**
  * Conan version: **conan 1.27.1**
  * Python version: **Python 3.7.5**

### Conan profile 
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=9
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]

```

### Steps to reproduce (Include if Applicable)

In conanfile.py
```
requires = (
...
        "sfml/2.5.1@bincrafters/stable",
```

Run `conan install ..`
cmake then fails with:

```
/usr/bin/ld: /home/xxx/.conan/data/libx11/1.6.8/bincrafters/stable/package/8cfe340971768a5f1c785526b65ae747644030c3/lib/libX11.a(OpenDis.o): undefined reference to symbol 'xcb_get_setup'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libxcb.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by charmyt** [comment]
Workaround. This is ugly. Is there a better way to temporarily fix this?

Edit the generated `link.txt` in the CMakeFiles dir.

Remove the X11 .a lib path pointing to the .conan cache.
`/home/xxx/.conan/data/libx11/1.6.8/bincrafters/stable/package/8cfe340971768a5f1c785526b65ae747644030c3/lib/libX11.a`

Replace it with the path to a local copy of the X11 .so e.g.
`/usr/lib/x86_64-linux-gnu/libX11.so.6`

**Comment by ericLemanissier** [comment]
I just updated `sfml/2.5.1@bincrafters/stable` so that it does not depend on `libx11/1.6.8@bincrafters/stable`, but on `xorg/system` which uses system-wide x11. Please retry with `-u` argument to update the recipe and tell us if this solves your problem. (please make sure you are using the latest version of conan)

**Comment by charmyt** [comment]
Damn that was a fast response. 

Yes this works now. Thank you. 

After applying your fix, on Ubuntu 19.10 I had to install the following:

```
sudo apt install libxcb-render-util0-dev
sudo apt install libxcb-xkb-dev
sudo apt install libxcb-icccm4-dev
sudo apt install libxcb-image0-dev
sudo apt install libxcb-keysyms1-dev
sudo apt install libxcb-xinerama0-dev
```

Not sure what the -u option is, I just rm'd my .conan cache and recreated my build dir. The X11 / xcb libraries seem to be linking correctly now.

Thanks again.
